types/raidgroupconfig_test.go

```diff
@@ -146,3 +146,160 @@ func TestPopulateDefaultGroupDeviceCount(t *testing.T) {
 		})
 	}
 }
+
+func TestGetDataDeviceCount(t *testing.T) {
+	var tests = map[string]struct {
+		src             *RaidGroupConfig
+		devicecount     []int64
```

Comment:
what is given & what is expectation here?

Reply:
Given, is total device count and expectation is how many are data devices.
For raidz there is one parity device, For raidz2 there is two parity device.

The main goal to get the device count to calculate capacity.
For mirror effective count is  - 1 (though there are 2 devices in the raid group)
For raidz effective count is  - 2^n (though there are 2^n + 1 devices in the raid group)
For raidz2 effective count is  - 2^n (though there are 2^n + 2 devices in the raid group)

Reply:
good. I was expecting the same naming conventions in the variables